### PR TITLE
When multiple classes are found by locate_class_in_module, find the one that is the deepest leaf of the desired subclass

### LINKED ISSUE #52: When Multiple settings files are found, pick the one that has the deepest leaf
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**
John Schock had a great solution that we'd like to move to

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by matthewfcarlson**
Fixes #52

### CODE REVIEW

File: edk2toollib/utility_functions.py [hunk 1]
```diff
@@ -392,30 +392,25 @@ def locate_class_in_module(Module, DesiredClass):
     there is a class defined in the module that subclasses DesiredClass.
```
Comment (corthon) [inline edk2toollib/utility_functions.py]: A quick ASCII art here would be useful.

File: edk2toollib/utility_functions.py [hunk 2]
```diff
@@ -392,30 +392,25 @@ def locate_class_in_module(Module, DesiredClass):
...
-    if len(DesiredClassInstances) == 0:  # we didn't find anything
-        return None
-    if len(DesiredClassInstances) > 1:  # we can't log because logging isn't setup yet
-        print(f"Multiple {DesiredClass.__name__} classes were found. Using {DesiredClassInstances[0].__name__}")
-    return DesiredClassInstances[0]  # return the first (highest priority class)
+            if (DesiredClassInstance is None) or issubclass(_class, DesiredClassInstance):
+                DesiredClassInstance = _class
+            elif not issubclass(DesiredClassInstance, _class):
+                raise RuntimeError(f"Multiple instances were found:\n\t{DesiredClassInstance}\n\t{_class}")
+    return DesiredClassInstance
```
Comment (corthon) [inline edk2toollib/utility_functions.py]: Is there any unit testing around this to ensure that correct classes are returned? This is the sort of thing that might blow up easily if Python were to change its behavior and we'd want to catch that ASAP.

File: edk2toollib/utility_functions.py (hunk 1 above)
Comment (matthewfcarlson) [reply edk2toollib/utility_functions.py]: I'll give it a shot

File: edk2toollib/utility_functions.py (hunk 2 above)
Comment (matthewfcarlson) [reply edk2toollib/utility_functions.py]: I'll write one

Comment (matthewfcarlson) [reply edk2toollib/utility_functions.py]: Added a test

File: edk2toollib/utility_functions.py (hunk 1 above)
Comment (matthewfcarlson) [reply edk2toollib/utility_functions.py]: Tried to add a quick something

**Pull request merged**